src/lib/background/session/storage.js

```diff
@@ -148,25 +148,60 @@ exports.removePage = removePage;
  * Update one or more tab page properties
  * @param {number} tabId
  * @param {Object} tabUpdate
- * @param {string} tabUpdate.title
- * @param {string} tabUpdate.url
+ * @param {string} [tabUpdate.title]
+ * @param {string} [tabUpdate.url]
  * @return {Promise}
  */
 function updatePage(tabId, tabUpdate)
 {
-  let {title, url} = tabUpdate;
-
-  let oldTabPage = tabPages.get(tabId);
-  if (!("url" in tabUpdate) && oldTabPage)
+  let {url} = tabUpdate;
+
+  let tabPage = {
+    attention: 0,
+    audible: false,
+    isAudio: false,
+    entity: null,
+    muted: false,
+    notification: null,
+    title: null,
+    url: null
+  };
+  tabPage = tabPages.get(tabId) || tabPage;
+
+  if (!("url" in tabUpdate))
   {
-    ({url} = oldTabPage);
+    ({url} = tabPage);
   }
 
   if (!url)
     return removePage(tabId);
```

Comment:
seems like we can rewrite this to be one conditional statement.

    if ("url" in tabUpdate)
    {
      // ... 
    }
    else
    {
      ({url} = tabPage);
      if (!url)
        return removePage(tabId);
    }